11-Backbonejs/Backbone-Views-of-Models-And-Events.md

```diff
@@ -1,103 +1,128 @@
 
 ## Adding our view to `app.js`
 
-We can update our renderList method to use the new view to draw each element.  Instead of calling the render method we created in `app.js` we create a new TaskView with the given model & template.
+We can update our code to use the new view to draw each element.  
 
-Then we render the template and append the resulting `$el` to the DOM.  
+Let's add our view to `app.js` with the following steps:
+1. Instead of calling the `render` method we created in `app.js`, let's create a new TaskView with the given model & template.
+2. Then let's render the template and append the resulting `$el` to the DOM.  
 
 ```javascript
 // app.js
 // imports etc...
 import TaskView from './views/task_view.js';
 
-var renderList = function(taskList) {
+const renderList = function(taskList) {
   // Clear the unordered list
-  $('.todo-items').empty();
-  
+  const $taskList = $('#todo-items');
+  $taskList.empty();
+
   // Iterate through the list rendering each Task
-  taskList.each(function(task) {
-  
+  taskList.forEach((task) => {
+
     // Create a new TaskView with the model & template
-    var taskView = new TaskView({
+    const taskView = new TaskView({
       model: task,
-      template: _.template($('#taskItemTemplate').html()),
-      tagName: 'li'
+      template: _.template($('#task-template').html()),
+      tagName: 'li',
+      className: 'task',
     });
-    
+
     // Then render the TaskView
     // And append the resulting HTML to the DOM.
-    $('.todo-items').append(taskView.render().$el);
+    $taskList.append(taskView.render().$el);
   });
 };
 ```
 
-Notice that in the code above we used `taskView.render().$el`.  We can do this because the view's `render()` method returns a reference to the view with the line `return this;`.  It is convention in Backbone to always have the `render()` method return `this` exactly so that we can do this kind of chaining.  We also set the 
-that we are using for `el` to be `li` or a list item.
+Notice that in the code above we used `taskView.render().$el`.  We can do this because the view's `render()` method returns a reference to the view with the line `return this;`.  It is convention in Backbone to always have the `render()` method return `this` exactly so that we can do this kind of chaining.  
+
+We also set the element that we are using for `el` to be `li`, such that our template will be wrapped in an `li` element. This will require us to adjust our template (`#task-template`) and remove the redundant `li`, and move the classes we had on that parent `li` to the `taskView` under the attribute className.
 
 We are performing many of the same operations we performed in our `app.js` file's `render` function as we create the `TaskView` and render it.
 
+Now that we've moved the rendering of a task to the TaskView, let's remove all references to `taskTemplate`, as they aren't being used anymore.
+
 ## DOM Events & Views
```

Comment:
The content in this section seems backwards.

For example, we tell the students to create an event handler for clicking the delete button, and at the end of that section mention that calling `remove` on the View also tells it to stop listening to events... but we've not shown any code for having the View listen to events. That code comes later on, in the "Backbone Events" section.

I think instead we should have a simpler, DOM-specific event example for this section and then move the delete button stuff into the "Views of Collections" textbook chapter, because destroying a Task model inherently involves the collection.

One suggestion for a DOM-specific event to use here instead is that maybe you can click the name of a task to highlight it as a priority, and just track that state as a property on the view. The more complete version of that, where it is stored as state on the model, is already covered by the example in "Backbone Events".

Reply:
Looking into reworking this section now... Just curious, why do you think that destroying a Task model inherently involves the collection? The TaskView has the delete button, and it can delete its own Task model, and on delete that will trigger an update on the TaskList that it's within. How do you see the TaskList being involved with that besides just updating/rerendering on change?